pkg/storage/chunk/cache/embeddedcache.go

```diff
@@ -161,40 +185,36 @@ func (c *EmbeddedCache) runPruneJob(interval, ttl time.Duration) {
 }
 
 // pruneExpiredItems prunes items in the cache that exceeded their ttl
-func (c *EmbeddedCache) pruneExpiredItems(ttl time.Duration) {
+func (c *EmbeddedCache[K, V]) pruneExpiredItems(ttl time.Duration) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
 	for k, v := range c.entries {
-		entry := v.Value.(*cacheEntry)
+		entry := v.Value.(*cacheEntry[K, V])
```

Comment:
Even though the cache value should not be of a different type, we should still check the cast result:

```suggestion
		entry, ok := v.Value.(*cacheEntry[K, V])
		if ok && time.Since(entry.updated) > ttl {
		    c.remove(k, v, expiredReason)
		}
```

Reply:
If the type is not expected, a panic seems appropriate